src/NameFieldSet.tsx

```diff
@@ -61,6 +61,8 @@ export class NameFieldSet extends React.Component<INameFieldSetProps, INameField
   }
 
   private setNames = () => {
+    const newNames = this.removeBlankNamesFromNameArray(this.state.names);
```

Comment:
you are not creating a new object wit this. You need to use the spread operator [... this.state.names]